pymisp/abstract.py

```diff
@@ -37,6 +39,53 @@ def tzname(self, dt):
         def dst(self, dt):
             return timedelta(0)
 
+    class MISPFileCache(object):
+        # cache up to 150 JSON structures in class attribute
+        __file_cache = cachetools.LFUCache(150)
+
+        @classmethod
+        def _load_json(cls, path):
+            # use root class attribute as global cache
+            file_cache = cls.__file_cache
+            # use modified time with path as cache key
+            mtime = os.path.getmtime(path)
+            if path in file_cache:
+                ctime, data = file_cache[path]
+                if ctime == mtime:
+                    return data
+            with open(path, 'rb') as f:
+                if OLD_PY3:
+                    data = json.loads(f.read().decode())
+                else:
+                    data = json.load(f)
+            file_cache[path] = (mtime, data)
+            return data
+
+else:
+    from collections.abc import MutableMapping
+    from functools import lru_cache
+    from pathlib import Path
+
+    resources_path = Path(__file__).parent / 'data'
+    misp_objects_path = resources_path / 'misp-objects' / 'objects'
+    with (resources_path / 'describeTypes.json').open('r') as f:
+        describe_types = json.load(f)['result']
+
+    class MISPFileCache(object):
+        # cache up to 150 JSON structures in class attribute
+
+        @classmethod
+        @lru_cache(maxsize=150)
```

Comment:
I think this should be `@staticmethod` to not include the cls parameter in the lru_cache key and have a global instead of a class-specific cache.

Reply:
Good point, changing that.